### PR TITLE
docs: add use description for specified result retry

### PR DESCRIPTION
**Pull request by YangruiEmma**
#### What type of PR is this?
docs

#### What this PR does / why we need it (en: English/zh: Chinese):
en: docs: add use description for specified result retry
zh: docs: 增加指定结果重试配置的说明

#### Which issue(s) this PR fixes:
none

### CODE REVIEW

File: content/zh/docs/kitex/Tutorials/basic-feature/retry.md [hunk 1]
```diff
@@ -88,6 +88,92 @@ fp.WithRetryBreaker(errRate float64)
 fp.WithRetrySameNode()
 ```
 
+###### 3.1.1.1 指定结果重试（异常/Resp）
+
+支持版本 v0.4.0。
+可配置支持指定结果重试，结果可以是请求失败，也可以指定 Resp。因为业务可能在 Resp 设置状态信息，针对某类返回重试，所以支持指定 Resp 重试，这里统称为异常重试。
```
Comment (GuangmingLuo) [inline content/zh/docs/kitex/Tutorials/basic-feature/retry.md:94]: 空行

File: content/zh/docs/kitex/Tutorials/basic-feature/retry.md [hunk 2]
```diff
@@ -88,6 +88,92 @@ fp.WithRetryBreaker(errRate float64)
...
+
+var opts []client.Option
+opts = append(opts, client.WithSpecifiedResultRetry(yourResultRetry))
+
+xxxCli := xxxservice.NewClient(targetService, opts...)
+```
+
+- retry.ShouldResultRetry 定义
+
+为了能具体到方法粒度对 error 和 resp 做判断，提供 rpcinfo 作为入参，可以通过 ri.To().Method() 获取方法。
```
Comment (GuangmingLuo) [inline content/zh/docs/kitex/Tutorials/basic-feature/retry.md:109]: 代码部分加一下 ``

File: content/zh/docs/kitex/Tutorials/basic-feature/retry.md [hunk 3]
```diff
@@ -88,6 +88,92 @@ fp.WithRetryBreaker(errRate float64)
...
+
+- 指定 异常/Resp 实现示例：
+
+  - 关于Resp：
+
+    Thrift 和 KitexProtobuf 协议 Resp 对应的是生成代码中的 *XXXResult，不是真实的业务 Resp，获取真实的 Resp 需要断言 interface{ GetResult() interface{} }；
+
+  - 关于 Error：
+
+    对端返回的 error，kitex 都会统一封装为 kerrors.ErrRemoteOrNetwork，对于 Thrift 和 KitexProtobuf 以下示例可以获取对端返回 Error Msg；对于 gRPC 如果对端通过 `status.Error` 构造的错误返回，本端使用 `status.FromError(err)` 可以获取 `*status.Status`，注意 `Status` 需使用 Kitex 提供的，包路径是 `github.com/cloudwego/kitex/pkg/remote/trans/nphttp2/status`。
```
Comment (GuangmingLuo) [inline content/zh/docs/kitex/Tutorials/basic-feature/retry.md:127]: 部分代码块没有加 ``

Comment (YangruiEmma) [reply content/zh/docs/kitex/Tutorials/basic-feature/retry.md]: done

**Review by GuangmingLuo — APPROVED** [review]

**Pull request merged**